### ISSUE TITLE
Traefik does not retry looking for Consul servers

### ISSUE THREAD

**Issue by davidgenn** [containous/traefik#2147]
### What did you do?
Start EC2 instance running Traefik and Consul client.

### What did you expect to see?
Traefik would poll Consul on specified url and retry if Consul hadn't started yet. 
This was working previously with v1.3.5 of Traefik.

### What did you see instead?
Traefik reported an error and did not retrieve Consul catalogue once Consul had started.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0-rc3
Codename:     roquefort
Go version:   go1.9
Built:        2017-09-18_04:38:27PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": true,
 "CheckNewVersion": true,
 "AccessLogsFile": "/var/log/traefik/access.log",
 "AccessLog": null,
 "TraefikLogsFile": "/var/log/traefik/traefik.log",
 "LogLevel": "DEBUG",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": {
  "ReadTimeout": 0,
  "WriteTimeout": 0,
  "IdleTimeout": 180000000000
 },
 "ForwardingTimeouts": {
  "DialTimeout": 30000000000,
  "ResponseHeaderTimeout": 0
 },
 "Docker": null,
 "File": null,
 "Web": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": {
  "Watch": false,
  "Filename": "",
  "Constraints": [],
  "Trace": false,
  "DebugLogGeneratedTemplate": false,
  "Endpoint": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "Domain": "service.consul",
  "ExposedByDefault": true,
  "Prefix": "traefik",
  "FrontEndRule": "Host:{{ServiceName}}.{{Domain}},PathPrefixStrip:/{{ServiceName}}"
 },
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": "/etc/traefik/traefik.toml"
}
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2017-09-20T06:02:25Z" level=info msg="Traefik version v1.4.0-rc3 built on 2017-09-18_04:38:27PM" 
time="2017-09-20T06:02:25Z" level=debug msg="Global configuration loaded {"GraceTimeOut":10000000000,"Debug":true,"CheckNewVersion":true,"AccessLogsFile":"/var/log/traefik/access.log","AccessLog":null,"TraefikLogsFile":"/var/log/traefik/traefik.log","LogLevel":"DEBUG","EntryPoints":{"http":{"Network":"","Address":":80","TLS":null,"Redirect":null,"Auth":null,"WhitelistSourceRange":null,"Compress":false,"ProxyProtocol":false}},"Cluster":null,"Constraints":[],"ACME":null,"DefaultEntryPoints":["http"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":0,"InsecureSkipVerify":false,"RootCAs":null,"Retry":null,"HealthCheck":{"Interval":30000000000},"RespondingTimeouts":{"ReadTimeout":0,"WriteTimeout":0,"IdleTimeout":180000000000},"ForwardingTimeouts":{"DialTimeout":30000000000,"ResponseHeaderTimeout":0},"Docker":null,"File":null,"Web":{"Address":":8080","CertFile":"","KeyFile":"","ReadOnly":false,"Statistics":null,"Metrics":null,"Path":"","Auth":null,"Debug":false,"CurrentConfigurations":null,"Stats":null,"StatsRecorder":null},"Marathon":null,"Consul":null,"ConsulCatalog":{"Watch":false,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"127.0.0.1:8500","Domain":"service.consul","ExposedByDefault":true,"Prefix":"traefik","FrontEndRule":"Host:{{ServiceName}}.{{Domain}},PathPrefixStrip:/{{ServiceName}}"},"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":null,"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null}" 
time="2017-09-20T06:02:25Z" level=info msg="Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:false} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s" 
time="2017-09-20T06:02:25Z" level=info msg="Starting provider *web.Provider {"Address":":8080","CertFile":"","KeyFile":"","ReadOnly":false,"Statistics":null,"Metrics":null,"Path":"","Auth":null,"Debug":true,"CurrentConfigurations":{},"Stats":{"Uptime":"2017-09-20T06:02:25.334560908Z","Pid":1416,"ResponseCounts":{},"TotalResponseCounts":{},"TotalResponseTime":"0001-01-01T00:00:00Z"},"StatsRecorder":null}" 
time="2017-09-20T06:02:25Z" level=info msg="Starting provider *consul.CatalogProvider {"Watch":false,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"127.0.0.1:8500","Domain":"service.consul","ExposedByDefault":true,"Prefix":"traefik","FrontEndRule":"Host:{{ServiceName}}.{{Domain}},PathPrefixStrip:/{{ServiceName}}"}" 
time="2017-09-20T06:02:25Z" level=info msg="Starting server on :80" 
time="2017-09-20T06:02:25Z" level=error msg="Failed to retrieve health checks" error="Unexpected response code: 500 (No known Consul servers)" 
time="2017-09-20T06:02:25Z" level=error msg="Failed to list services: Unexpected response code: 500 (No known Consul servers)" 
time="2017-09-20T06:02:55Z" level=warning msg="Error checking new version: Get https://update.traefik.io/repos/containous/traefik/releases: dial tcp 104.27.184.74:443: i/o timeout" 
```

**Comment by SaikiranDaripelli** [comment]
Will it be part of 1.4? i am facing same issue.

**Comment by mmatur** [comment]
@SaikiranDaripelli  and @davidgenn could you retry with this image `mmatur/traefik:consul_retry`

**Comment by SaikiranDaripelli** [comment]
@mmatur I tried above image and i am getting below error if consul is not available on startup with consul KV configuration and traefik container is exiting.
`traefik_1  | 2017/10/16 06:13:17 Error loading configuration: Get http://192.168.1.71:8500/v1/kv/traefik?recurse=&wait=30000ms: dial tcp 192.168.1.71:8500: getsockopt: connection refused
ptinfra_traefik_1 exited with code 255`

Whereas consulCatelog is working fine  as expected when consul is down on startup and brought up later.

**Comment by emilevauge** [comment]
@SaikiranDaripelli should be fixed in https://github.com/containous/traefik/pull/2268

**Comment by traefiker** [comment]
Closed by #2263.
